### ISSUE TITLE
Mat-file 7.3 support  

### ISSUE THREAD

**Issue by geofjamg** [HebiRobotics/MFL#67]
Any plan to support mat-file version 7.3?

**Comment by ennerf** [comment]
I briefly looked into it, but the Java libraries for HDF seemed pretty awful to use and there is barely any documentation on the libraries and the format. We currently also don't have an internal use case for it, so I can't justify pursuing this further. Sorry.

We are open for PRs though in case anyone else wants to get it supported. The APIs were originally created with >2GB indexing in mind.

**Comment by geofjamg** [comment]
Indeed,  this would be a huge work to support it. Maybe an intermediate support would be to just clearly wam the user about file is a mat-file 7.3 and that it is not yet supported. With current code we only get a "This is not a valid MATLAB 5.0 MAT-file description." which has not been meaningful enough for our users. I think this become more and more important because Matlab seems in last versions to save mat file in 7.3 by default. With a clear exception like "This is a Mat-file 7.3 is it not yet supported, supported versions are from X to  7", it would be more clear for a user to just re-save the file in version 7 using the correct flag.

**Comment by geofjamg** [comment]
But this is not an issue related to your lib. The API is clear and is only related to Mat 5 format.
Thanks for your answer.

**Comment by ennerf** [comment]
I just did a quick test, and `2022a` still seems to default to `-v7 `. 

I changed the error message to something [more descriptive](https://github.com/HebiRobotics/MFL/blob/develop/mfl-core/src/main/java/us/hebi/matlab/mat/format/Mat5File.java#L296-L297): `This file was saved using the unsupported MATLAB 7.3 MAT-file version ('-v7.3' flag). Only files saved with the '-v7' version flag can be loaded.`

I also just came across [jHDF](https://github.com/jamesmudd/jhdf) which wasn't around last time I checked.

**Comment by ennerf** [comment]
I've released the better error messages as `0.5.9`, so I'll go ahead and close this issue.

It looks like the HDF data is structured in a similar manner to the proprietary format. It's still not something I have time for right now, but a better HDF library makes the problem a lot more tractable.

* `sparse.mat` as `-v7`

```
spa = 5x6 sparse
```

* `sparse.mat` as `-v7.3`

```
Group [name=spa, path=/spa/, address=0x320] (attributes: MATLAB_class=double,MATLAB_sparse=5,)
  data: DatasetBase [path=/spa/data] (attributes: ) (io.jhdf.object.datatype.FloatingPoint@1e67b872, double)
  ir: DatasetBase [path=/spa/ir] (attributes: ) (io.jhdf.object.datatype.FixedPoint@60addb54, BigInteger)
  jc: DatasetBase [path=/spa/jc] (attributes: ) (io.jhdf.object.datatype.FixedPoint@3f2a3a5, BigInteger)
```

**Comment by geofjamg** [comment]
Thanks!
